src/flowgentic/langGraph/memory.py

```diff
@@ -605,28 +666,171 @@ class MemoryEnabledState(BaseModel):
+					self._checkpointer = DragonCheckpointSaver(
+						managers_per_node=self.config.managers_per_node,
+						n_nodes=self.config.n_nodes,
+						total_mem=self.config.checkpoint_total_mem,
+					)
+					logger.info("Initialized DragonCheckpointSaver for checkpoints")
+
+			except Exception as e:
+				logger.warning(f"Failed to initialize Dragon backends: {e}")
+				if self.config.fallback_to_local:
+					logger.info("Falling back to local memory storage")
+					self._store = None
+					self._checkpointer = None
+				else:
+					raise
+		else:
+			logger.info("Using local memory storage (distributed mode not enabled)")
+
+		self._initialized = True
+
+	def get_store(self):
+		"""Get the distributed store backend.
+
+		Returns:
+			DragonDDictStore instance or None if not available/enabled
+		"""
+		self._initialize_backends()
+		return self._store
+
+	def get_checkpointer(self):
+		"""Get the distributed checkpoint saver backend.
+
+		Returns:
+			DragonCheckpointSaver instance or None if not available/enabled
+		"""
+		self._initialize_backends()
+		return self._checkpointer
+
+	def is_distributed(self) -> bool:
+		"""Check if distributed memory is actually being used.
+
+		Returns:
+			True if using distributed backends, False if using local
+		"""
+		self._initialize_backends()
+		return self._store is not None or self._checkpointer is not None
+
+	def get_memory_info(self) -> Dict[str, Any]:
+		"""Get information about the current memory configuration.
+
+		Returns:
+			Dictionary with memory backend information
+		"""
+		self._initialize_backends()
+
+		return {
+			"mode": self.config.mode,
+			"is_distributed": self.is_distributed(),
+			"store_enabled": self._store is not None,
+			"checkpointer_enabled": self._checkpointer is not None,
+			"n_nodes": self.config.n_nodes,
+			"managers_per_node": self.config.managers_per_node,
+			"store_memory_gb": self.config.store_total_mem / (1024**3),
+			"checkpoint_memory_gb": self.config.checkpoint_total_mem / (1024**3),
+		}
+
+	def close(self):
+		"""Clean up distributed memory resources."""
+		if self._store and hasattr(self._store, "close"):
+			try:
+				self._store.close()
+				logger.info("Closed distributed store")
+			except Exception as e:
+				logger.warning(f"Error closing distributed store: {e}")
+
+		if self._checkpointer and hasattr(self._checkpointer, "close"):
+			try:
+				self._checkpointer.close()
+				logger.info("Closed distributed checkpointer")
+			except Exception as e:
+				logger.warning(f"Error closing distributed checkpointer: {e}")
+
+
 # Facade class following the established pattern for LangGraph integration
 class LangraphMemoryManager:
 	"""Facade for memory management in LangGraph integration.
 
 	Follows the same pattern as LangraphToolFaultTolerance and AgentLogger,
 	providing a simple interface for memory operations within the LangraphIntegration.
+
+	Supports both local (short-term) and distributed (long-term) memory backends.
```

Comment:
Distributed and long-term are two different things. Can you point me to where we do the long-term approach? (Sorry if I missed it in this PR)